### PR TITLE
HPCC-11835 New Topology

### PR DESCRIPTION
**Pull request by GordonSmith**
Reworking of Topology for new ECLWatch.

Signed-off-by: Gordon Smith gordon.smith@lexisnexis.com

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please Review

File: esp/src/eclwatch/ESPTopology.js [hunk 1]
```diff
@@ -51,6 +59,581 @@ define([
...
+                    return "Sasha";
+            }
+            return "Unknown";
+        },
+
+        addServices: function (items) {
+            arrayUtil.forEach(items, function (item) {
+                //  Push Bindings/LogDirectory into machines  ---
+                /*
+                if (lang.exists("TpMachines.TpMachine", item) {
```
Comment (miguelvazq) [inline esp/src/eclwatch/ESPTopology.js]: remove commented if (lang.exists(

File: esp/src/eclwatch/TopologyWidget.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+############################################################################## */
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
```
Comment (miguelvazq) [inline esp/src/eclwatch/TopologyWidget.js]: array not used

File: esp/src/eclwatch/TopologyWidget.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    "dojo/on",
+
+    "dijit/registry",
+    "dijit/form/Button",
+    "dijit/form/ToggleButton",
+    "dijit/ToolbarSeparator",
+    "dijit/layout/ContentPane",
+
+    "dgrid/selector",
```
Comment (miguelvazq) [inline esp/src/eclwatch/TopologyWidget.js]: selector not used

File: esp/src/eclwatch/TopologyWidget.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
+    "dojo/on",
+
+    "dijit/registry",
+    "dijit/form/Button",
```
Comment (miguelvazq) [inline esp/src/eclwatch/TopologyWidget.js]: button not used

File: esp/src/eclwatch/TopologyWidget.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+############################################################################## */
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
```
Comment (miguelvazq) [inline esp/src/eclwatch/TopologyWidget.js]: lang not used

**Comment by miguelvazq** [comment]
Review complete. See my comments above.

**Comment by GordonSmith** [comment]
Changes made as per comments.

**Comment by miguelvazq** [comment]
Looks good.

**Pull request merged**